tests/bcl-test/BCLTests/TestRunner.Core/TcpTextWriter.cs

```diff
@@ -0,0 +1,97 @@
+﻿// this is an adaptation of NUnitLite's TcpWriter.cs with an additional 
+// overrides and with network-activity UI enhancement
+// This code is a small modification of 
+// https://github.com/spouliot/Touch.Unit/blob/master/NUnitLite/TouchRunner/TcpTextWriter.cs
+using System;
+using System.IO;
+using System.Net.Sockets;
+using System.Text;
+
+using UIKit;
+
+namespace BCLTests.TestRunner.Core {
+	public class TcpTextWriter : TextWriter {
+		
+		TcpClient client;
+		StreamWriter writer;
+
+		public TcpTextWriter (string hostName, int port)
+		{
+			if ((port < 0) || (port > ushort.MaxValue))
+				throw new ArgumentException ("port");
```

Comment:
`throw new ArgumentOutOfRangeException (nameof (port), $"Port must be between 0 and {ushort.MaxValue}" );`